contracts/EmergencyExecutionMultisig.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import {RestrictedMultisigBase} from "./RestrictedMultisigBase.sol";
+
+interface IEmergencyProtectedTimelock {
+    function emergencyExecute(uint256 proposalId) external;
+    function emergencyReset() external;
+}
+
+contract EmergencyExecutionMultisig is RestrictedMultisigBase {
+    uint256 public constant EXECUTE_PROPOSAL = 1;
+    uint256 public constant RESET_GOVERNANCE = 2;
+
+    address emergencyProtectedTimelock;
+
+    constructor(
+        address _owner,
+        address[] memory _members,
+        uint256 _quorum,
+        address _emergencyProtectedTimelock
+    ) RestrictedMultisigBase(_owner, _members, _quorum) {
+        emergencyProtectedTimelock = _emergencyProtectedTimelock;
+    }
+
+    // Proposal Execution
+    function voteExecuteProposal(uint256 _proposalId, bool _supports) public onlyMember {
+        _vote(_buildExecuteProposalAction(_proposalId), _supports);
+    }
+
+    function getExecuteProposalState(uint256 _proposalId)
+        public
+        returns (uint256 support, uint256 ExecutionQuorum, bool isExecuted)
+    {
+        return _getState(_buildExecuteProposalAction(_proposalId));
+    }
+
+    function executeProposal(uint256 _proposalId) public {
+        _execute(_buildExecuteProposalAction(_proposalId));
+    }
+
+    // Governance reset
+
+    function voteGoveranaceReset() public onlyMember {
```

Comment:
i'd rename this to `approveGovernanceReset` (see the comment for `approveEmergencyActivate`)